optuna/distributions.py

```diff
@@ -84,14 +84,15 @@ def _asdict(self):
     def __eq__(self, other):
         # type: (Any) -> bool
 
-        if not isinstance(other, type(self)):
-            return False
-
+        if not type(self) is type(other):
```

Comment:
Changed to evaluate different distribution types as non-equal objects (c.f. existing `test_distributions.py`).

Reply:
As the name implies, `NotImplemented` constant means that the comparison with `other` is not supported. If you want to treats the different distributions as "non-equal objects", I think it's better to return `False`.

Reply:
Thanks for the input. You have a point, we're defining the equality here. Let me change the implementation.

Reply:
You are welcome. But sorry, I mean you should return `False` in the comparison between distribution classes like `UniformDistribution() == LogUniformDistribution()`. So it still need to return `NotImplemented` constant in the comparison like `UniformDistribution() == Foo()`.

I think it's better to insert `if not isinsntance(other, BaseDistribution): return NotImplemented` before this line.

Reply:
That makes total sense, thanks. I changed the implementation accordingly.